sentry-kotlin-multiplatform-gradle-plugin/src/main/java/io/sentry/kotlin/multiplatform/gradle/FrameworkPathResolver.kt

```diff
@@ -0,0 +1,256 @@
+            when {
+                basePath.endsWith("Sentry.xcframework") -> FrameworkPaths.createValidated(
+                    staticBasePath = basePath,
+                    architectures = architectures
+                )
+
+                basePath.endsWith("Sentry-Dynamic.xcframework") -> FrameworkPaths.createValidated(
+                    dynamicBasePath = basePath,
+                    architectures = architectures
+                )
+
+                else -> FrameworkPaths.NONE
+            }
+        } ?: FrameworkPaths.NONE
+        if (linker.frameworkPath.orNull != null && result == FrameworkPaths.NONE) {
+            project.logger.warn(
+                "Custom framework path has been set manually but could not be found. " +
+                    "Trying to resolve framework paths using other strategies."
+            )
+        }
+        return result
+    }
+}
+
+/**
+ * Finds framework paths based on the derived data path.
+ *
+ * This strategy prioritizes:
+ * 1. A user-specified Xcode project path via [LinkerExtension].
+ * 2. An auto-found Xcode project in the root directory. (mainly works for mono repo)
+ */
+class DerivedDataStrategy(
+    private val project: Project,
+    private val derivedDataProvider: (String) -> String? = { xcodeprojPath ->
+        project.providers.of(DerivedDataPathValueSource::class.java) {
+            it.parameters.xcodeprojPath.set(xcodeprojPath)
+        }.orNull
+    }
+) : FrameworkResolutionStrategy {
+    private val linker: LinkerExtension = project.extensions.getByType(LinkerExtension::class.java)
+
+    override fun resolvePaths(architectures: Set<String>): FrameworkPaths {
+        val xcodeprojSetByUser = linker.xcodeprojPath.orNull?.takeIf { it.isNotEmpty() }
+        val foundXcodeproj = xcodeprojSetByUser ?: findXcodeprojFile(project.rootDir)?.absolutePath
+        if (foundXcodeproj == null) {
+            return FrameworkPaths.NONE
+        }
+
+        val derivedDataPath = derivedDataProvider(foundXcodeproj) ?: return FrameworkPaths.NONE
+        val dynamicBasePath =
+            "$derivedDataPath/SourcePackages/artifacts/sentry-cocoa/Sentry-Dynamic/Sentry-Dynamic.xcframework"
+        val staticBasePath =
+            "$derivedDataPath/SourcePackages/artifacts/sentry-cocoa/Sentry/Sentry.xcframework"
+
+        return FrameworkPaths.createValidated(
+            dynamicBasePath = dynamicBasePath,
+            staticBasePath = staticBasePath,
+            architectures = architectures
+        )
+    }
+
+    /**
+     * Searches for a xcodeproj starting from the root directory. This function will only work for
+     * monorepos and if it is not, the user needs to provide the [LinkerExtension.xcodeprojPath].
+     */
+    private fun findXcodeprojFile(startingDir: File): File? {
+        val ignoredDirectories = listOf("build", "DerivedData")
+
+        fun searchDirectory(directory: File): File? {
+            val files = directory.listFiles() ?: return null
```

Comment:
I guess we could also use `walkFileTree` or something like that from kotlin, but this is also fine like it's done now

Reply:
ah yeah thats a better api to use I think, I'll have a look at it

Reply:
changed to using `walkFileTree` instead